### DISCUSSION TITLE
Websocket Connection Lost

### DISCUSSION THREAD

**Discussion by jbreske** [forkineye/ESPixelStick#665] [Q&A]
Have Espixelstick v4 loaded on four Dig Quad AE+.  I am firing them back up since last Christmas.  When opening Espixelstick via the IP address, one of the four will momentarily connect then give me a "Websocket Connection Lost...Attempting to Reconnect".  I reflashed the chip and getting the same error.  Any ideas?  It never stays connected.  The other three I can access just fine.

**Comment by MartinMueller2003** [comment]
It is obvious you did not use the newest version when you reflashed. I have removed the websocket functionality because it was causing issues and using a ton of ram. 
Keep the ESP connected to the flash tool. The tool will track the log output. After a failure, save the log and attach it to this conversation.

**Reply by jbreske** [reply]
Thanks. I definitely didn't pull the latest. It was the beta 4 version available around last November. I was comfortable with what I ran last Christmas on the four Dig Quads so didn't go looking for newer. I'll pull a log for you and also pull the latest build. I appreciate the quick response.

**Comment by jbreske** [comment]
Assuming I generated the log correctly.  I left my ESP attached to the USB data cable.  Left the esp flashtool open.  Accessed the ESP via the IP address.  Received the Websocket Connection Lost message.  Saved the log from the esp flashtool. Log attached.  Thanks.
[JBreske Websocket Log](https://github.com/forkineye/ESPixelStick/files/12667589/JBreske.Websocket.Log)

**Comment by MartinMueller2003** [comment]
Here is the  problem: The last message is WS Client disconnect. The latest version does NOT HAVE WEBSOCKETs. That means you are NOT running the latest code. Websockets are gone because they are not stable and cause memory exhaustion which would make a system look like it is offline.

**Reply by jbreske** [reply]
Thanks for the diagnosis. I pulled the beta 4 last November (not sure of the version) so I could use the benefit of FPP and the Dig Quad ae+ SD card for my light show over the holiday. What is the stable or beta version I can flash that will allow the FPP usage on the dig quad with ae+? I started firing my four dig quads back up now in preps for halloween. This is one of four that gave me this problem. The other 3 are fine so I was thinking I had a hardware problem. I definitely don't know what a websocket is or does. Thanks for the assist.

**Comment by MartinMueller2003** [comment]
Grab the CI builds from August 22 or later. They have the latest stable code.

**Comment by MartinMueller2003** [comment]
FYI: Websockets are no longer used by the ESP V4 code.
